x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/PutInferenceModelAction.java

```diff
@@ -70,6 +81,33 @@ public BytesReference getContent() {
             return content;
         }
 
+        public BytesReference getRewrittenContent() {
```

Comment:
Can you clarify what the purpose of this function is? My understanding is we should be setting the endpoint version either based on the users provided endpoint version or based on the default endpoint version if none is provided. Why are we setting the endpoint version for the user? If the endpoint version is just based off of the users input, do we even have a reason to store it? Shouldn't we just parse any input and decide on our end where it's stored in the Model object?

Reply:
The idea here is that we don't want the user to interact with the endpoint version directly. They won't set the endpoint version or see it (looks like I missed preventing the user from setting it). The endpoint versioning will be handled behind the scenes in a transparent way, the user can use either parameters or task settings, and they don't have to worry about the endpoint version. 

I think you're right about not needing the endpointVersion for the code as I've written it. I think we only need endpointVersion if we only want to return one of `parameters` or `task_settings` at any time, but we might just be better off always returning both.